crates/services/p2p/src/service.rs

```diff
@@ -115,6 +115,10 @@ pub enum TaskRequest {
         channel: OnResponse<Option<Vec<SealedBlockHeader>>>,
     },
     GetTransactions {
+        block_height_range: Range<u32>,
+        channel: OnResponse<Option<Vec<Transactions>>>,
```

Comment:
The problem that I have with this solution is that in case of the p2p that select the peer I want p2p to return the peer so that I can report things about his results afterwards. Something that I don't need when I give the peer because I already know it.